__tests__/commands/FQLXCommands.test.js

```diff
@@ -0,0 +1,630 @@
+const FQLXCommands = require("../../commands/FQLXCommands");
+const {Client, fql} = require("fauna");
+const clientConfig = require("../config")
+const Logger = require("../../Logger");
+
+
+describe('FQLXCommands', () => {
+  let faunaClient
+  const log = jest.fn()
+  const logger = new Logger({ log })
+
+  const cleanup = async () => {
+    let res = await faunaClient.query(fql`Function.all().map(f => f.delete())`)
+    let a = res.data?.after
+    while (a != null) {
+      const page = await faunaClient.query(fql`Set.paginate(${a})`)
+      a = page.data?.after
+    }
+  }
+
+  const objToArray = (obj) => {
+
+    return Object.entries(obj).map(([k, v]) => {
+      return {
+        name: k,
+        ...v,
+      }
+    })
+  }
+
+  const mergeMetadata = (f) => {
+    const md = {created_by_serverless_plugin: "fauna:v10", deletion_policy: "destroy" }
+    const merged = { ...md, ...f.data}
+    return {
+      ...f,
+      data: merged,
+    }
+  }
+
+  const verify = async ({ funcs = []}) => {
+    funcs = funcs.map(mergeMetadata).sort((a, b) => a.name > b.name ? 1 : -1)
+
+    const actual = await faunaClient.query(fql`Function.all().order( .name )`)
+    if (funcs.length !== actual.data.data.length) {
+      expect(actual.data.data).toEqual(funcs)
+    }
+
+    for (let i = 0; i < funcs.length; i++) {
+      const e = funcs[i]
+      const a = actual.data.data[i]
+
+      expect(a.data).toEqual(e.data)
+      expect(a.name).toEqual(e.name)
+      expect(a.body).toEqual(e.body)
+      expect(a.role).toEqual(e.role)
+    }
+  }
+
+  beforeAll(async () => {
+    const p = clientConfig.port
+    const ep = `${clientConfig.scheme}://${clientConfig.domain}${p ? ":" + p : ""}`
+    faunaClient = new Client({
+      secret: clientConfig.secret,
+      endpoint: new URL(ep),
+    })
```

Comment:
why make a new client every time? it seems like this would just make tests take longer.